pkg/client/e2e.go

```diff
@@ -32,14 +34,25 @@ import (
 func (*SonobuoyClient) GetTests(reader io.Reader, show string) ([]reporters.JUnitTestCase, error) {
 	read := results.NewReaderWithVersion(reader, "irrelevant")
 	junitResults := reporters.JUnitTestSuite{}
-	err := read.WalkFiles(func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
-		// TODO(chuckha) consider reusing this function for any generic e2e-esque plugin results.
-		// TODO(chuckha) consider using path.Join()
-		return results.ExtractFileIntoStruct(results.PluginsDir+e2e.ResultsSubdirectory+e2e.JUnitResultsFile, path, info, &junitResults)
-	})
+	e2eJunitPath := path.Join(results.PluginsDir, e2e.ResultsSubdirectory, e2e.JUnitResultsFile)
+
+	found := false
+	err := read.WalkFiles(
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			// TODO(chuckha) consider reusing this function for any generic e2e-esque plugin results.
+			// TODO(chuckha) consider using path.Join()
+			if path == e2eJunitPath {
+				found = true
+			}
+			return results.ExtractFileIntoStruct(e2eJunitPath, path, info, &junitResults)
+		})
+	if !found {
```

Comment:
Should we check the error first to see if the read failed, just so it's clear that the file was found, but it errored in some way.

Reply:
If it wasn't found then it wouldn't have failed when reading; it will never try to read the file before setting `found=true`

Reply:
I'm thinking broader if some error happens and `err` is returned. We're checking if `!found` first before dealing with the possible error that gets returned. 

Maybe I'm not reading the code correctly on how the flow works.

Reply:
Yea, its a function thats just being passed to the file-walker.

So the function itself takes an error so that when you look at file-2 you can check if the walker failed when looking at file-1. Here we just stop at the first error ever encountered.

In addition, if you look at the `ExtractFileIntoStruct` it just does nothing unless the path of the current file matches the one you're looking for.

I could make it more clear by by just stating that if the paths arent the same to just return nil. It wouldn't be functionally different than it is now but it wouldn't require you to know how the `ExtractFileIntoStruct` is implemented. 👍

Reply:
Here's the flow I'm seeing:

1. `found == false` (as default)
2. `read.WalkFiles(func())` returns an error
3. Line 53 checks if `found == false` which it is since that's the default. The code then tells the user that it `failed to find results file` when really the file could be there, but extracting or something else returned an error. 
4. err isn't checked until line 58

Reply:
:dunce:

Thanks for walking me through it! I was looking at the other `err` and thinking that's what you meant. I hadn't even noticed the other one because it was so far away!

For some reason the original writers wanted to define the slice even if they were going to return an empty one. I just return a nil slice and its fine so I rightfully moved it up higher in the code flow.

I updated the tests to confirm the proper code flow in each of them. It makes the tests brittle because of tying the message into the test, but I don't think its a terrible burden to carry for now :)